src/layoutBuilder.js

```diff
@@ -823,20 +1131,38 @@ LayoutBuilder.prototype.buildNextLine = function (textNode) {
 		isForceContinue = inline.noNewLine && !isHardWrap;
 	}
 
+	// The inlineRtl flag is passed through automatically. Only run the inline version
+	// of the function if rtl isn't set, but inlineRtl is (to avoid breaking changes)
+	if (!textNode.rtl && line.inlines.some((inline) => inline.inlineRtl)) {
+		// This code allows us to copy the existing styleStack (so we can use inline styles to
+		// change the fonts or word styling)
+		const styleStack = this.docMeasure.styleStack.clone();
+		styleStack.push(textNode);
+		const availableWidth = this.writer.context().availableWidth;
+		return addLineWithInlineRTL(
+			line,
+			styleStack,
+			textTools,
+			textNode,
+			availableWidth
+		);
+	}
```

Comment:
The third change required to call the new function when the `inlineRtl` flag is set.